www/js/lib/filecache.js

```diff
@@ -0,0 +1,187 @@
+     */
+    LRUCache.prototype.get = function (id) {
+        var entry = this._entries[id];
+        if (entry === undefined) {
+            return entry;
+        }
+        this.moveToTop(entry);
+        return entry.value;
+    };
+
+    /**
+     * Stores a value in the cache by id and prunes the least recently used entry if the cache is larger than MAX_CACHE_SIZE
+     * @param {String} id The key under which to store the value (consists of filename + file number)
+     * @param {Uint16Array} value The value to store in the cache 
+     */
+    LRUCache.prototype.store = function (id, value) {
+        var entry = this._entries[id];
+        if (entry === undefined) {
+            entry = this._entries[id] = { id: id, prev: null, next: null, value: value };
+            this.insertAtTop(entry);
+            if (this._size >= this._limit) {
+                var e = this._last;
+                this.unlink(e);
+                delete this._entries[e.id];
+            } else {
+                this._size++;
+            }
+        } else {
+            entry.value = value;
+            this.moveToTop(entry);
+        }
+    };
+
+    /**
+     * Delete a cache entry
+     * @param {String} entry The entry to delete 
+     */
+    LRUCache.prototype.unlink = function (entry) {
+        if (entry.next === null) {
+            this._last = entry.prev;
+        } else {
+            entry.next.prev = entry.prev;
+        }
+        if (entry.prev === null) {
+            this._first = null;
+        } else {
+            entry.prev.next = entry.next;
+        }
+    };
+
+    /**
+     * Insert a cache entry at the top of the cache
+     * @param {String} entry The entry to insert 
+     */
+    LRUCache.prototype.insertAtTop = function (entry) {
+        if (this._first === null) {
+            this._first = this._last = entry;
+        } else {
+            this._first.prev = entry;
+            entry.next = this._first;
+            this._first = entry;
+        }
+    };
+
+    /**
+     * Move a cache entry to the top of the cache
+     * @param {String} entry The entry to move 
+     */
+    LRUCache.prototype.moveToTop = function (entry) {
+        this.unlink(entry);
+        this.insertAtTop(entry);
+    };
+
+    // Create a new cache
+    var cache = new LRUCache(MAX_CACHE_SIZE);
+
+    // Counters for reporting only
+    var hits = 0;
+    var misses = 0;
+
+    /**
+     * Read a certain byte range in the given file, breaking the range into chunks that go through the cache
+     * If a read of more than blocksize (bytes) is requested, do not use the cache
+     * @param {Object} file The requested ZIM archive to read from
+     * @param {Integer} begin The byte from which to start reading
+     * @param {Integer} end The byte at which to stop reading (end will not be read)
+     * @return {Promise<Uint8Array>} A Promise that resolves to the correctly concatenated data from the split ZIM file set
```

Comment:
Change description to "A Promise that resolves to the correctly concatenated data from the cache or from the ZIM archive". (This function is now indifferent as to whether the ZIM is split or not, as that is handled in `zimfile.js`.)